extensions/amp-date-picker/1.0/range-date-picker.js

```diff
@@ -0,0 +1,256 @@
+import {addDays, differenceInDays, isAfter, isSameDay} from 'date-fns';
+import {DayPicker} from 'react-day-picker';
+
+import {
+  closestAncestorElementBySelector,
+  scopedQuerySelector,
+} from '#core/dom/query';
+
+import * as Preact from '#preact';
+import {useCallback, useEffect, useMemo, useRef, useState} from '#preact';
+import {ContainWrapper} from '#preact/component';
+
+import {
+  DateFieldNameByType,
+  DateFieldType,
+  DatePickerMode,
+  FORM_INPUT_SELECTOR,
+  TAG,
+} from './constants';
+import {getFormattedDate, parseDate} from './date-helpers';
+import {DayButton} from './day-button';
+
+/**
+ * @param {!DateInput.Props} props
+ * @return {PreactDef.Renderable}
+ */
+export function RangeDatePicker({
+  allowBlockedEndDate,
+  allowBlockedRanges,
+  blockedDates,
+  children,
+  endInputSelector,
+  format,
+  id,
+  initialVisibleMonth,
+  mode,
+  onError,
+  startInputSelector,
+}) {
+  const [startDate, setStartDate] = useState();
+  const [endDate, setEndDate] = useState();
+
+  const [startInputAttributes, setStartInputAttributes] = useState();
+  const [endInputAttributes, setEndInputAttributes] = useState();
+
+  const onErrorRef = useRef(onError);
+  const containerRef = useRef();
+
+  const [isOpen] = useState(mode === DatePickerMode.STATIC);
+
+  const [showChildren, setShowChildren] = useState(true);
```

Comment:
I'm sure there's a better way to handle this, but I'm not quite sure what it is yet!

Basically, the user has the choice to either pass in their own input field, or the component will create a hidden input field for them. 

If there is an input field, we will want to get the element using `scopedQuerySelector` and read its attributes and value. If this input field exists, we want to either turn it into or replace it with our controlled input. Right now, I'm creating new inputs based on what the user passes in, and hiding the children once we have created new inputs to avoid showing duplicate inputs. Is anyone aware of a better solution here?